plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/SpanAddingClassVisitorFactory.kt

```diff
@@ -69,17 +78,38 @@ abstract class SpanAddingClassVisitorFactory :
             )
             SentryPlugin.logger.info { "Read sentry-android sdk state: $sdkState" }
             val instrumentables = listOfNotNull(
-                AndroidXSQLiteDatabase().takeIf { sdkState.isAtLeast(PERFORMANCE) },
-                AndroidXSQLiteStatement().takeIf { sdkState.isAtLeast(PERFORMANCE) },
-                AndroidXRoomDao().takeIf { sdkState.isAtLeast(PERFORMANCE) },
+                AndroidXSQLiteDatabase().takeIf {
+                    isDatabaseInstrEnabled(sdkState, parameters.get())
+                },
+                AndroidXSQLiteStatement().takeIf {
+                    isDatabaseInstrEnabled(sdkState, parameters.get())
+                },
+                AndroidXRoomDao().takeIf { isDatabaseInstrEnabled(sdkState, parameters.get()) },
                 ChainedInstrumentable(
                     listOf(WrappingInstrumentable(), RemappingInstrumentable())
-                ).takeIf { sdkState.isAtLeast(FILE_IO) }
+                ).takeIf { isFileIOInstrEnabled(sdkState, parameters.get()) }
             )
+            SentryPlugin.logger.debug {
+                "Instrumentables: ${instrumentables.joinToString { it::class.java.simpleName }}"
+            }
             parameters.get()._instrumentables = ArrayList(instrumentables)
             return instrumentables
         }
 
+    private fun isDatabaseInstrEnabled(
+        sdkState: SentryAndroidSdkState,
+        parameters: SpanAddingParameters
+    ): Boolean =
+        sdkState.isAtLeast(PERFORMANCE) &&
+            parameters.features.get().contains(InstrumentationFeature.DATABASE)
+
+    private fun isFileIOInstrEnabled(
+        sdkState: SentryAndroidSdkState,
+        parameters: SpanAddingParameters
+    ): Boolean =
+        sdkState.isAtLeast(FILE_IO) &&
```

Comment:
`m`: Why isn't this `PERFORMANCE`, but `FILE_IO`?

Reply:
bcoz we are relying on some runtime classes which are available only from a certain version (FILE_IO=5.5.0 in this case), otherwise we would crash at runtime